server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -500,6 +506,58 @@ private void startLookups(List<LookupBean> lookupBeanList)
     }
   }
 
+  /**
+   * @return a pair of the map with successful lookups and the list of failed lookups
+   */
+  private Pair<Map<LookupBean, LookupExtractorFactoryContainer>, List<LookupBean>> startLookups(
+      List<LookupBean> lookupBeans,
+      CompletionService<Map.Entry<LookupBean, LookupExtractorFactoryContainer>> completionService,
+      List<Future<?>> allFutures
+  ) throws InterruptedException
+  {
+    for (LookupBean lookupBean : lookupBeans) {
+      allFutures.add(completionService.submit(() -> startLookup(lookupBean)));
+    }
+    Map<LookupBean, LookupExtractorFactoryContainer> successfulLookups = new HashMap<>();
+    for (int i = 0; i < lookupBeans.size(); i++) {
+      Future<Map.Entry<LookupBean, LookupExtractorFactoryContainer>> completedFuture = completionService.take();
+      try {
+        Map.Entry<LookupBean, LookupExtractorFactoryContainer> lookupResult = completedFuture.get();
+        if (lookupResult != null) {
+          successfulLookups.put(lookupResult.getKey(), lookupResult.getValue());
+        }
+      }
+      catch (ExecutionException e) {
+        // not adding to successfulLookups
+      }
+    }
+    List<LookupBean> failedLookups = lookupBeans
+        .stream()
+        .filter(l -> !successfulLookups.containsKey(l))
+        .collect(Collectors.toList());
+    return new Pair<>(successfulLookups, failedLookups);
+  }
+
+  @Nullable
+  private Map.Entry<LookupBean, LookupExtractorFactoryContainer> startLookup(LookupBean lookupBean)
+  {
+    LookupExtractorFactoryContainer container = lookupBean.getContainer();
+    LOG.info("Starting lookup [%s]:[%s]", lookupBean.getName(), container);
+    try {
+      if (container.getLookupExtractorFactory().start()) {
+        LOG.info("Started lookup [%s]:[%s]", lookupBean.getName(), container);
+        return new AbstractMap.SimpleImmutableEntry<>(lookupBean, container);
+      } else {
+        LOG.error("Failed to start lookup [%s]:[%s]", lookupBean.getName(), container);
+        return null;
+      }
+    }
+    catch (Exception e) {
+      LOG.error(e, "Failed to start lookup [%s]:[%s]", lookupBean.getName(), container);
```

Comment:
If this throws an error it should not need to log it. Perhaps you can 

`throw new RE(e, "Failed to start lookup [%s]:[%s]", lookupBean.getName(), container);`

Reply:
Changed